src/EvaluateAsync.cpp

```diff
@@ -0,0 +1,54 @@
+#include "EvaluateAsync.h"
+#include "WebPage.h"
+#include "WebPageManager.h"
+#include "JsonSerializer.h"
+#include <iostream>
+
+EvaluateAsync::EvaluateAsync(WebPageManager *manager, QStringList &arguments, QObject *parent) : SocketCommand(manager, arguments, parent) {
+}
+
+void EvaluateAsync::start() {
+  QString script = arguments()[0];
+  QString jsonArgs;
+  if (arguments().length()>1){
+    jsonArgs = arguments()[1];
+  } else {
+    jsonArgs ="[]";
+  }
+  QString eval_script = QString("(function(){"
```

Comment:
There's a `capybara.js` file that's injected onto every page. Is it possible to add this function in there?

Reply:
Probably, but that wasn't how src/Evaluate.cpp was written and I'm not 100% sure whether there are any special permissions allowed for JS scripts loaded by Qt (as opposed to scripts loaded by the visited page).  If there are special permissions then having an accessible `eval` may not be great.

Reply:
Ah, right - we have to inject this differently because we're interpolating JavaScript code in there.